tools/caffe.cpp

```diff
@@ -34,6 +34,13 @@ DEFINE_string(solver, "",
     "The solver definition protocol buffer text file.");
 DEFINE_string(model, "",
     "The model definition protocol buffer text file.");
+DEFINE_string(phase, "",
+    "Optional; network phase (TRAIN or TEST). Only used for 'time'.");
+DEFINE_int32(level, 0,
```

Comment:
That is what I had in mind, but perhaps that's counterintuitive too and we should instead catch when these args are given to `caffe train` and complain. It's reasonable to not support overriding, and my concern is more about silently running a different net than intended.